paasta_tools/tron_tools.py

```diff
@@ -234,13 +242,106 @@ def __init__(
+                system_paasta_config
+            ),
+            "spark.kubernetes.namespace": SPARK_KUBERNETES_NAMESPACE,
+            "spark.kubernetes.executor.label.yelp.com/paasta_service": truncated_service,
+            "spark.kubernetes.executor.label.yelp.com/paasta_instance": truncated_instance,
+            "spark.kubernetes.executor.label.yelp.com/paasta_cluster": self.get_cluster(),
+            "spark.kubernetes.executor.label.paasta.yelp.com/service": truncated_service,
+            "spark.kubernetes.executor.label.paasta.yelp.com/instance": truncated_instance,
+            "spark.kubernetes.executor.label.paasta.yelp.com/cluster": self.get_cluster(),
+            "spark.kubernetes.node.selector.yelp.com/pool": self.get_pool(),
+            "spark.kubernetes.executor.label.yelp.com/pool": self.get_pool(),
+        }
+
+        if self.get_team() is not None:
+            conf["spark.kubernetes.executor.label.yelp.com/owner"] = (self.get_team(),)
+
+        # Spark defaults to using the Service Account that the driver uses for executors,
+        # but that has more permissions than what we'd like to give the executors, so use
+        # the default service account instead
+        conf["spark.kubernetes.authenticate.executor.serviceAccountName"] = "default"
+        if self.get_iam_role() and self.get_iam_role_provider() == "aws":
+            conf[
+                "spark.kubernetes.authenticate.executor.serviceAccountName"
+            ] = create_or_find_service_account_name(
+                iam_role=self.get_iam_role(),
+                namespace=EXECUTOR_TYPE_TO_NAMESPACE[self.get_executor()],
+                dry_run=self.for_validation,
+            )
+
+        conf.update(
+            setup_volume_mounts(
+                volumes=self.get_volumes(
+                    system_volumes=system_paasta_config.get_volumes(),
+                ),
+            ),
+        )
+
+        # most of the service_configuration_lib function expected string values only
+        # so let's go ahead and convert the values now instead of once per-wrapper
+        strigified_spark_args = {
+            k: (str(v) if not isinstance(v, bool) else str(v).lower())
+            for k, v in spark_args.items()
+        }
+        # Core ML has some translation logic to go from Mesos->k8s options for people living in the past
+        conf.update(
+            adjust_spark_resources(
+                spark_args=strigified_spark_args, desired_pool=self.get_pool()
+            )
+        )
+        # as well as some QoL tweaks for other options
+        conf.update(setup_shuffle_partitions(spark_args=strigified_spark_args))
+        conf.update(setup_event_log_configuration(spark_args=strigified_spark_args))
+        # now that we've added all the required stuff, we can add in all the stuff that users have added
+        # themselves
+        conf.update(_filter_user_spark_opts(user_spark_opts=strigified_spark_args))
```

Comment:
service_configuration_lib did this before we adjusted for event_log/shuffle_partitions/resources; is this intentionally changing it so user-defined options override any that aren't `NON_CONFIGURABLE_SPARK_OPTS` ? 

e.g. would user-specified resources here remove the extra room we've given ourselves in adjust_spark_resources?

Reply:
ah, you're right - thanks for catching that! I think we wanna do this at the very beginning and then `.update()` into that dict

Reply:
(this is also what i get for being bad at writing tests - i would have caught this if i had paid more attention to the conf we're asserting in tests :p)

Reply:
updated this :)